### PR TITLE
Use pyproject.toml

### PR DESCRIPTION
**Pull request by lkollar**
Make setuptools a build dependency in pyproject.toml and set it as the
build backend.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[build-system]
+requires = ["setuptools", "pbr"]
```
Comment (pganssle) [inline pyproject.toml]: Does this need `wheel` as well?

File: tests/test_manylinux.py [hunk 2]
```diff
@@ -195,7 +195,7 @@ def test_build_wheel_with_binary_executable(docker_container):
     policy, manylinux_id, python_id, io_folder = docker_container
     docker_exec(manylinux_id, 'yum install -y gsl-devel')
 
-    docker_exec(manylinux_id, ['bash', '-c', 'cd /auditwheel_src/tests/testpackage && python setup.py bdist_wheel -d /io'])
+    docker_exec(manylinux_id, ['bash', '-c', 'cd /auditwheel_src/tests/testpackage && python -m pip wheel -w /io .'])
```
Comment (pganssle) [inline tests/test_manylinux.py]: If you are going to use `pip` to build wheels, it's best to also use `--no-deps`.

I personally prefer `pep517.build`.

Comment (lkollar) [reply tests/test_manylinux.py]: As this is only invoked for for a test inside a container, I'd leave `pip` here to keep it simple. Adding `--no-deps`.

**Comment by lkollar** [comment]
@auvipy given that this is a relatively small change and has already been approved by Paul, I'm going to land it.

**Pull request merged**